From 87a23c03d053cb490eab5e631fae9cf80ce17974 Mon Sep 17 00:00:00 2001
From: Aditya Kumar Singh <quic_adisi@quicinc.com>
Date: Mon, 16 Oct 2023 15:26:56 +0530
Subject: [PATCH] wifi: nl80211: allow CAC while operation is going on in
 case of MLO

Commit "ff311bc11a7e: (nl80211: allow CAC only if no operation is going on)"
added logic to return -EBUSY while starting radar detection if operation is
already going on the intended interface. However, with MLO, it is possible
that interface is operating already and one of the link is starting radar
detection. Hence, same check does not apply to MLO.

Add change to not check for operation in case of MLO.

Signed-off-by: Aditya Kumar Singh <quic_adisi@quicinc.com>
---
 net/wireless/nl80211.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/wireless/nl80211.c b/net/wireless/nl80211.c
index 917b228c4ead..718a18d03cd9 100644
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -10633,7 +10633,7 @@ static int nl80211_start_radar_detection(struct sk_buff *skb,
 		goto unlock;
 	}
 
-	if (netif_carrier_ok(dev)) {
+	if (!wdev->valid_links && netif_carrier_ok(dev)) {
 		err = -EBUSY;
 		goto unlock;
 	}
-- 
2.17.1

